test/integration/scripting_spec.mjs

```diff
@@ -1771,28 +1772,52 @@ describe("Interaction", () => {
 
   describe("in autoprint.pdf", () => {
     let pages;
+    const printHandles = new Map();
 
     beforeAll(async () => {
       // Autoprinting is triggered by the `Open` event, which is one of the
       // first events to be dispatched to the sandbox, even before scripting
       // is reported to be ready. It's therefore important that `loadAndWait`
-      // returns control as soon as possible after opening the PDF document,
-      // and the first element we can check for is the `<html>` tag of the
-      // viewer. Note that the `autoprint.pdf` file is very small, so printing
+      // returns control as soon as possible after opening the PDF document.
+      // Note that the `autoprint.pdf` file is very small, so printing
       // it is usually very fast and therefore activating the selector check
       // too late will cause it to never resolve because printing is already
       // done (and the printed page div removed) before we even get to it.
-      pages = await loadAndWait("autoprint.pdf", "html");
+      pages = await loadAndWait(
+        "autoprint.pdf",
+        "",
+        null /* pageSetup = */,
+        async page => {
+          printHandles.set(
+            page,
+            await page.evaluateHandle(() => [
+              new Promise(resolve => {
+                globalThis.printResolve = resolve;
+              }),
+            ])
+          );
+          await page.waitForFunction(() => {
+            if (!window.PDFViewerApplication?.eventBus) {
+              return false;
+            }
+            window.PDFViewerApplication.eventBus._on("print", () => {
+              globalThis.printResolve();
+            });
```

Comment:
Should this use `once: true` to ensure that the listener is being removed?

Also, outside of the `web/`-folder you want to use `eventBus.on(...)` since the method used here is intended to only be "internal".